### ISSUE TITLE
MacOS fatal error: too many errors emitted, stopping now [-ferror-limit=] while trying to make

### ISSUE THREAD

**Issue by SreehariRamMohan** [CMU-Perceptual-Computing-Lab/openpose#809]
### Posting rules
1. **Fill** the **Your System Configuration section (all of it!)** if you have some kind of error or performance question.
2. **No** questions about **training**. OpenPose only implements testing.
3. **No** questions about **3rd party libraries**.
    - Caffe errors/issues, check [Caffe](http://caffe.berkeleyvision.org) documentation.
    - CUDA check failed errors: They are usually fixed by re-installing CUDA, then re-installing the proper cuDNN version, and then re-compiling (or re-installing) OpenPose. Otherwise, check for help in CUDA forums.
    - OpenCV errors: Install the default/pre-compiled OpenCV or check for online help.
4. **No duplicated** posts.
5. **No** posts about **questions already answered / clearly explained in** the **documentation** (e.g. **no more low-speed nor out-of-memory questions**).
6. Set a **proper issue title**: add the Ubuntu/Windows word and be specific (e.g. do not simple call it: `Compile error`).
7. Only English comments.
Issues/comments which do not follow these rules will be **ignored or removed** with no further clarification.

### Issue Summary

Whenever I run **make -j`nproc`** I get the following errors
`fatal error: too many errors emitted, stopping now [-ferror-limit=]
45 warnings and 20 errors generated.
make[5]: *** [src/caffe/CMakeFiles/caffeproto.dir/__/__/include/caffe/proto/caffe.pb.cc.o] Error 1
make[4]: *** [src/caffe/CMakeFiles/caffeproto.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [caffe/src/openpose_caffe-stamp/openpose_caffe-build] Error 2
make[1]: *** [CMakeFiles/openpose_caffe.dir/all] Error 2
make: *** [all] Error 2
`

### Executed Command (if any)
make -j`nproc`

### OpenPose Output (if any)

`fatal error: too many errors emitted, stopping now [-ferror-limit=]
45 warnings and 20 errors generated.
make[5]: *** [src/caffe/CMakeFiles/caffeproto.dir/__/__/include/caffe/proto/caffe.pb.cc.o] Error 1
make[4]: *** [src/caffe/CMakeFiles/caffeproto.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [caffe/src/openpose_caffe-stamp/openpose_caffe-build] Error 2
make[1]: *** [CMakeFiles/openpose_caffe.dir/all] Error 2
make: *** [all] Error 2
`

### Your System Configuration
1. **OpenPose version**:
latest version from `git clone https://github.com/CMU-Perceptual-Computing-Lab/openpose
`

2. **General configuration**:
    - **Installation mode**: build from source
    - **Operating system** : macOS High Sierra 10.13.6 
    - **Release or Debug mode**? : release
    - **Compiler** `Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
`

3. **Non-default settings**:
    - **3-D Reconstruction module added**? no

4. **3rd-party software**:
    - **Caffe version**: Default from OpenPose, custom version, ...? default from openpose
    - **CMake version** : cmake version 3.12.1

    - **OpenCV version**:  compiled from source 

6. If **CPU-only mode** issue:
    - **CPU brand & model**: 2 GHz Intel Core i5
    - Total **RAM memory** available: 8 gb

7. If **Python** API:
    - **Python version**:  Python 3.6.4 :: Anaconda custom (64-bit)
    - **Numpy version** 1.14.0

**Comment by soulslicer** [comment]
Please paste the error to pastebin and then paste the link here

**Comment by SreehariRamMohan** [comment]
@soulslicer Point noted. 

Here is the pastebin link of the full error log. 

https://pastebin.com/ywbRhKeE

**Comment by carlovirtucio** [comment]
@SreehariRamMohan I'm using MacOS High Sierra as well. I assume you got the same issue with CMake and cloned caffe directly from git?

I got it to work by following this: https://stackoverflow.com/a/31010221

Basically I just added `set(CMAKE_CXX_STANDARD 11)` to the following files:
`openpose/build/caffe/src/openpose_caffe-build/CMakeFiles/3.12.1/CMakeCXXCompiler.cmake`
`openpose/build/CMakeFiles/3.12.1/CMakeCXXCompiler.cmake`

**Comment by soulslicer** [comment]
This is a protobuf issue. Either there is some unknown bug of caffe with OSX 10.13, or you have Anaconda installed which has a incompatible version of protobuf. But it looks like its the latter

**Comment by SreehariRamMohan** [comment]
@soulslicer Yes you are correct I have installed Anaconda (which is why I think there is an incompatible version of protobuf). How can I solve this issue? Some ideas I have for solving the issue are to upgrade the version of protobuf in anaconda (but which version should I download), or to create a virtual environment (not entirely sure how to do this). What course of action would you recommend to allow me to download openpose successfully?

**Comment by SreehariRamMohan** [comment]
@carlovirtucio will this fix the issue with my incompatible protobuf version from Anaconda?

**Comment by carlovirtucio** [comment]
@SreehariRamMohan I’m not sure if it would. Have you tried the steps with your machine? (I can detail it out more if needed)

**Comment by SreehariRamMohan** [comment]
@carlovirtucio I did the steps that you suggested (using vim to add set(CMAKE_CXX_STANDARD 11) at the top of both files) however I still get the same error.

**Comment by carlovirtucio** [comment]
@SreehariRamMohan Hmm I guess it really is the `protobuf` issue with Anaconda. The possible reason why it worked for me is probably because I don't have `conda` installed in my device.

**Comment by SreehariRamMohan** [comment]
How do you suggest I fix this issue @gineshidalgo99

**Comment by SreehariRamMohan** [comment]
@carlovirtucio  What else can I try

**Comment by soulslicer** [comment]
Please take a look at the other issues. Unfortunately, installing anaconda installs a protobuf that is incompatible with Caffe. Either you uninstall anaconda and install protobuf via apt-get, or you compile your own caffe and link it to openpose

https://github.com/CMU-Perceptual-Computing-Lab/openpose/issues/677

Please see above
